src/account.rs

```diff
@@ -61,15 +65,38 @@ where
         }
     }
 
-    fn pair(&self) -> &P {
+    fn pair(&self) -> Result<P, P::DeriveError> {
         match self {
-            Self::Root { pair, .. } => pair,
-            Self::Sub { .. } => todo!(),
+            Self::Root { pair, .. } => Ok(pair.clone()),
+
+            Self::Sub { path, .. } => self.maybe_pair(path.clone())
+        }
+    }
+
+    //TODO: change to use regex
+    fn maybe_pair(&self, path: String) -> Result<P, P::DeriveError> {
+        let p: Vec<&str> = path.split("/").collect();
+        match self {
+            Self::Root { pair, ..} => pair.derive(p[1 .. ].into_iter().map(|s| DeriveJunction::from(s)).into_iter(), None).map(|a| a.0),
+            Self::Sub {parent, .. } => {
+                match parent.pair() {
```

Comment:
```suggestion
                let parent = parent.pair()?;
```
when dealing with `Result` and `Option` there's usually a more idiomatic way to extract or change the data than `match`ing.

Reply:
Changed in commit https://github.com/virto-network/libwallet/pull/15/commits/c9c87195f2b85e057a462597afb8e8432856542c